test/builder-api/test.sh

```diff
@@ -1,34 +1,282 @@
+
+[superuser]
+name = 'travis'
+password = 'travis'
+EOF
+
+  if [ -d "$pg_svc_dir" ]; then
+    echo "$pg_svc_dir exists. Moving user.toml into place."
+    sudo mv $dir/user.toml $pg_svc_dir
+  else
+    echo "$pg_svc_dir still doesn't exist. WTF. Expect more failures."
+  fi
+fi
+
+# This will produce a URI that looks like
+# postgresql://hab@127.0.0.1:39605/test
+pg_url=$(sudo su $user -c "pg_tmp -t -w 240 -o \"-c max_locks_per_transaction=128\"")
+port=$(echo "$pg_url" | awk -F ":" '{ print $3 }' | awk -F "/" '{ print $1 }')
+
+# Write out some config files
+cat << EOF > $dir/config_api.toml
+[depot]
+builds_enabled = true
+non_core_builds_enabled = true
+key_dir = "$key_dir"
+
+[github]
+app_private_key = "$key_dir/builder-github-app.pem"
+
+[segment]
+write_key = "hahano"
+EOF
+
+cat << EOF > $dir/config_jobsrv.toml
+key_dir = "$key_dir"
+
+[archive]
+backend = "local"
+local_dir = "/tmp"
+
+[datastore]
+host = "127.0.0.1"
+port = $port
+user = "$user"
+database = "test"
+connection_retry_ms = 300
+connection_timeout_sec = 3600
+connection_test = false
+pool_size = 8
+EOF
+
+cat << EOF > $dir/config_sessionsrv.toml
+[permissions]
+admin_team = 1995301
+build_worker_teams = [2555389]
+early_access_teams = [1995301]
+
+[github]
+app_private_key = "$key_dir/builder-github-app.pem"
+
+[datastore]
+host = "127.0.0.1"
+port = $port
+user = "$user"
+database = "test"
+connection_retry_ms = 300
+connection_timeout_sec = 3600
+connection_test = false
+pool_size = 8
+EOF
+
+cat << EOF > $dir/config_worker.toml
+auth_token = "bobo"
+bldr_url = "http://localhost:9636"
+auto_publish = true
+data_path = "/tmp"
+
+[github]
+app_private_key = "$key_dir/builder-github-app.pem"
+EOF
+
+cat << EOF > $dir/config_originsrv.toml
+[datastore]
+host = "127.0.0.1"
+port = $port
+user = "$user"
+database = "test"
+connection_retry_ms = 300
+connection_timeout_sec = 3600
+connection_test = false
+pool_size = 8
+EOF
+
+cat << EOF > $dir/Procfile
+api: $base_dir/target/debug/bldr-api start --path $dir/depot --config $dir/config_api.toml
+router: $base_dir/target/debug/bldr-router start
+jobsrv: $base_dir/support/run-server jobsrv $dir/config_jobsrv.toml
+sessionsrv: $base_dir/support/run-server sessionsrv $dir/config_sessionsrv.toml
+originsrv: $base_dir/support/run-server originsrv $dir/config_originsrv.toml
+worker: $base_dir/target/debug/bldr-worker start --config $dir/config_worker.toml
+EOF
+
+cat << EOF > $dir/bldr.env
+RUST_LOG=debug
+RUST_BACKTRACE=1
+HAB_DOCKER_STUDIO_IMAGE="habitat-docker-registry.bintray.io/studio"
+EOF
+
+# Start all the services up
+if [ "$platform" = "mac" ]; then
+  echo "Running these tests on a mac"
+  env HAB_FUNC_TEST=1 "$base_dir/support/mac/bin/forego" start -f "$dir/Procfile" -e "$dir/bldr.env" > "$dir/services.log" 2>&1 &
+else
+  echo "Running these tests on linux"
+  env HAB_FUNC_TEST=1 "$base_dir/support/linux/bin/forego" start -f "$dir/Procfile" -e "$dir/bldr.env" > "$dir/services.log" 2>&1 &
+fi
+
 forego_pid=$!
```

Comment:
If the `forego` command is launched with `sudo`, here's an approach that will allow the shutdown at the end to work correctly:
```bash
sudo_pid=$!
while pid_is_running $sudo_pid && [[ -z ${forego_pid:-} ]]; do
  forego_pid=$(pgrep -P "$sudo_pid")
done
```